artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -154,13 +154,17 @@ public void addScanAck(BooleanSupplier retryBeforeScan, ToIntFunction<PagedRefer
 
    @Override
    public void performScanAck() {
-      // we should only have a max of 2 scheduled tasks
-      // one that's might still be currently running, and another one lined up
-      // no need for more than that
-      if (scheduledScanCount.incrementAndGet() <= 2) {
-         executor.execute(this::actualScanAck);
-      } else {
-         scheduledScanCount.decrementAndGet();
+      // notice performScanAck is supposed to be called from a single thread.
```

Comment:
Might be useful for the API to say this if its supposed to, and perhaps which thread that is expected to be.